### PR TITLE
Override default gitleaks configuration

### PR DESCRIPTION
**Pull request by ivankalitaonefootball**
## Override default gitleaks configuration
We use [default](https://github.com/zricethezav/gitleaks/blob/master/config/default.go) gitleaks rules. These rules sometimes have some flaws, e.g. we have a lot of false positives for the Twitter Client IDs because of the default Twitter rules ([more information](https://github.com/zricethezav/gitleaks/pull/614)). To fix this problem I override the default gitleaks rules with the rules provided by this repository (`/gitleaks.toml`).

Also, I set a reasonable default (`.of/security/gitleaks.toml`) for the `config-path` input argument and fix its description to reflect the fact that it's not the main configuration file but an additional config.

### Questions
- How does the release process for this action work? If I understand correctly we'll need to either override the tag `v1.6.0-of-custom` or update all the repositories to use the new version of motain/gitleaks-action
- Is it ok that one of the gitleaks actions fails? I mean should I fix it? If I do this the fork will diverge too much from the upstream.
- I'm not sure how much OF semantics/processes I can share in this repository because it's public.

### CODE REVIEW

File: .gitleaks.toml [hunk 1]
```diff
@@ -1,106 +0,0 @@
-title = "gitleaks config"
```
Comment (ivankalitaonefootball) [inline .gitleaks.toml:1]: This file was not used at all 🤷

File: action.yml [hunk 2]
```diff
@@ -5,9 +5,9 @@ branding:
   icon: eye
 inputs:
   config-path:
-    description: 'Path to config (relative to $GITHUB_WORKSPACE)'
+    description: 'Path to additional configuration (relative to $GITHUB_WORKSPACE)'
     required: false
-    default: '.github/.gitleaks.toml'
```
Comment (ivankalitaonefootball) [inline action.yml:10]: This path does not exist on our repositories, instead, we use `.of/security/gitleaks.toml`, so I changed it.

File: gitleaks.toml [hunk 3]
```diff
@@ -0,0 +1,174 @@
+title = "gitleaks config"
```
Comment (ivankalitaonefootball) [inline gitleaks.toml:1]: https://github.com/zricethezav/gitleaks/blob/8533e76ffc4824cfadffa4996c0e6330bd0b0509/config/default.go

**Comment by andreabenfatto** [comment]
Closing because there was some misinterpretation of how we wanted to use Gitleaks,

**Pull request closed without merging**